### PR TITLE
[core] Layer manager is responsible for buckets and layouts creation

### PR DESCRIPTION
**Pull request by pozdnyakov**
This is a step to stop using of render layers in tile worker thread.

### CODE REVIEW

**Comment by pozdnyakov** [comment]
Tag https://github.com/mapbox/mapbox-gl-native/issues/13812

File: src/mbgl/layermanager/layer_factory.cpp [hunk 1]
```diff
@@ -21,6 +25,16 @@ optional<std::string> LayerFactory::getSource(const style::conversion::Convertib
     return source;
 }
 
+std::unique_ptr<Bucket> LayerFactory::createBucket(const BucketParameters&, const std::vector<const RenderLayer*>&) noexcept {
+    assert(false);
```
Comment (ivovandongen) [inline src/mbgl/layermanager/layer_factory.cpp]: Why not pure virtual?

File: src/mbgl/layermanager/layer_factory.cpp [hunk 2]
```diff
@@ -21,6 +25,16 @@ optional<std::string> LayerFactory::getSource(const style::conversion::Convertib
...
+    return nullptr;
+}
+
+std::unique_ptr<Layout> LayerFactory::createLayout(const LayoutParameters&, std::unique_ptr<GeometryTileLayer>, const std::vector<const RenderLayer*>&) noexcept {
+    assert(false);
```
Comment (ivovandongen) [inline src/mbgl/layermanager/layer_factory.cpp]: idem

File: src/mbgl/layermanager/layer_factory.cpp (hunk 1 above)
Comment (pozdnyakov) [reply src/mbgl/layermanager/layer_factory.cpp]: Did not want to repeat the same empty definitions for the concrete factories that do not use this method.

File: src/mbgl/layermanager/layer_factory.cpp (hunk 2 above)
Comment (pozdnyakov) [reply src/mbgl/layermanager/layer_factory.cpp]: Ditto.

**Pull request merged**